tools/ksvalidator.py

```diff
@@ -80,42 +81,62 @@ def main(argv):
-    except KickstartDeprecationWarning as err:
-        return (cleanup(destdir, processedFile),
-                [_("File uses a deprecated option or command.\n%s") % err])
-    except KickstartParseError as err:
-        return (cleanup(destdir, processedFile), [str(err)])
-    except KickstartError:
-        return (cleanup(destdir, processedFile),
-                [_("General kickstart error in input file")])
-    except Exception as e:                                      # pylint: disable=broad-except
-        return (cleanup(destdir, processedFile),
-                [_("General error in input file:  %s") % e])
+    retval = 0
+    retmsg = []
+
+    # unpack any globs
+    ksfiles = []
+    for inksfile in opts.ksfile:
+        if is_url(inksfile):
+            ksfiles.append(inksfile)
+        else:
+            ksfiles.extend(glob.glob(inksfile))
+
+    # check if there are any files to check
+    if not ksfiles:
+        return (1, "No files match the patterns.")
+
+    # iterate over files to check them
+    with tempfile.TemporaryDirectory(prefix="ksvalidator-tmp-") as destdir:
+        for ksfile in ksfiles:
+            print(_("Checking kickstart file %(filename)s\n" % {"filename": ksfile}))
+
+            try:
+                f = load_to_file(ksfile, os.path.join(destdir, "ks.cfg"))
+            except KickstartError as e:
+                retval += 1
+                retmsg.append(_("Error reading %(filename)s:\n%(version)s") % {"filename": ksfile, "version": e})
+
+            try:
+                handler = makeVersion(opts.version)
+            except KickstartVersionError:
+                # return immediately because bad version is fatal for all files
+                return (1, [_("The version %s is not supported by pykickstart") % opts.version])
+
+            # turn kickstart parse warnings into errors
+            warnings.filterwarnings(action="error", category=KickstartParseWarning)
+
+            ksparser = KickstartParser(handler, followIncludes=opts.followincludes,
+                                       errorsAreFatal=opts.firsterror)
+
+            try:
+                processedFile = preprocessKickstart(f)
+                ksparser.readKickstart(processedFile)
+                retval += ksparser.errorsCount
+            except KickstartDeprecationWarning as err:
+                retval += 1
+                retmsg.append(_("File uses a deprecated option or command.\n%s") % err)
+            except KickstartParseError as err:
+                retval += 1
+                retmsg.append(str(err))
+            except KickstartError:
+                retval += 1
+                retmsg.append(_("General kickstart error in input file"))
+            except Exception as e:        # pylint: disable=broad-except
+                retval += 1
+                retmsg.append(_("General error in input file:  %s") % e)
+
+    return retval, retmsg
 
 if __name__ == "__main__":
     retval, messages = main(sys.argv[1:])
```

Comment:
Pylint doesn't like retval here and in main, rename it to 'rc' or similar.

Reply:
Done.